src/wallet/scriptpubkeyman.cpp

```diff
@@ -732,12 +787,27 @@ bool LegacyScriptPubKeyMan::AddWatchOnly(const CScript& dest, int64_t nCreateTim
 void LegacyScriptPubKeyMan::SetHDChain(const CHDChain& chain, bool memonly)
 {
     LOCK(cs_wallet);
-    if (!memonly && !WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain))
-        throw std::runtime_error(std::string(__func__) + ": writing chain failed");
+    if (!memonly) {
+        if (!WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain)) {
```

Comment:
I'm new to Bitcoin and I'm still learning the testing standards - should there be a test where this exception is expected to be thrown?

Reply:
No. This is only expected to be thrown under corruption and/or hardware failure conditions.